spell-check.yaml

```
@@ -0,0 +1,22 @@
+name: Check spelling
+
+on:
+  workflow_call:
+
+jobs:
+  spellcheck:
```

[Nits] Let's make the naming consistent.

```suggestion
  spell-check:
```